### DISCUSSION TITLE
Mapping edge_attr back to original node indices after sampling

### DISCUSSION THREAD

**Discussion by jasperhyp** [pyg-team/pytorch_geometric#4423] [Q&A]
I sampled a huge graph and got a small `Data` object for each batch, but I cannot find an efficient way to map `edge_attr` back to the original node indices after I map `edge_index` back to the original node indices with the following code:

```python
ori_map = batch.n_id
batch_edge_index = ori_map[batch.edge_index]
```

where `n_id` is initially just `torch.arange(num_of_nodes)`.

The only way I can think of is to directly search for each new edge in the mapped edge indices...  But that is way too inefficient.

**Comment by rusty1s** [comment]
Can't you do
```python
data.e_id = torch.arange(data.num_edges)
...
data.edge_attr[batch.e_id] = batch.edge_attr
```

**Reply by jasperhyp** [reply]
Ah sorry my bad, that works.  I tried this before but I came across other issues, then I forgot this orz

**Reply by jasperhyp** [reply]
I think `e_id` won't work because we'll still need to call the original-length `edge_index` to get mapping.

But actually this is a dumb question...  We don't need to map `edge_attr` back because once we've mapped edge_index back edge_attr is just working as usual orz  Sorry for taking your time!
